node/src/blockchain/blockchain_bridge.rs

```diff
@@ -1517,128 +1810,114 @@ mod tests {
                 client_id: 1234,
                 context_id: 4321,
             }),
         };
-
-        subject.handle_scan(
-            success_handler,
-            ScanType::Receivables,
-            retrieve_transactions,
-        );
-
-        System::current().stop();
-        system.run();
-        let accountant_recording = accountant_recording_arc.lock().unwrap();
-        assert_eq!(accountant_recording.len(), 0);
-    }
-
-    #[test]
-    fn handle_scan_handles_failure_without_skeleton() {
-        init_test_logging();
-        let (accountant, _, accountant_recording_arc) = make_recorder();
+        let blockchain_transaction = BlockchainTransaction {
+            block_number: start_block,
+            from: wallet,
+            wei_amount: 20_000,
+        };
+        let retrieved_blockchain_transactions = RetrievedBlockchainTransactions {
+            new_start_block: start_block,
+            transactions: vec![blockchain_transaction],
+        };
         let mut subject = BlockchainBridge::new(
-            Box::new(BlockchainInterfaceMock::default()),
-            Box::new(PersistentConfigurationMock::new()),
+            Box::new(
+                BlockchainInterfaceMock::default()
+                    .retrieve_transactions_result(Ok(retrieved_blockchain_transactions)),
+            ),
+            Box::new(persistent_config),
             false,
         );
         let system = System::new("test");
-        subject.scan_error_subs_opt = Some(accountant.start().recipient());
-        let retrieve_transactions = RetrieveTransactions {
-            recipient: make_wallet("somewallet"),
-            response_skeleton_opt: None,
-        };
-
-        subject.handle_scan(
-            failure_handler,
+        let accountant_addr = accountant
+            .system_stop_conditions(match_every_type_id!(ScanError))
+            .start();
+        subject.received_payments_subs_opt = Some(accountant_addr.clone().recipient());
+        subject.scan_error_subs_opt = Some(accountant_addr.recipient());
+        subject.handle_scan_future(
+            BlockchainBridge::handle_retrieve_transactions,
             ScanType::Receivables,
             retrieve_transactions,
         );
 
-        System::current().stop();
         system.run();
         let accountant_recording = accountant_recording_arc.lock().unwrap();
-        let message = accountant_recording.get_record::<ScanError>(0);
-        assert_eq!(
-            message,
-            &ScanError {
-                scan_type: ScanType::Receivables,
-                response_skeleton_opt: None,
-                msg: "My tummy hurts".to_string()
-            }
-        );
-        assert_eq!(accountant_recording.len(), 1);
-        TestLogHandler::new().exists_log_containing("WARN: BlockchainBridge: My tummy hurts");
+        let msg_opt = accountant_recording.get_record_opt::<ScanError>(0);
+        assert_eq!(msg_opt, None, "We didnt expect a scan error: {:?}", msg_opt);
```

Comment:
**URGENCY STATUS: LOW, BUT YOU'VE CREATED A CONFUSING TEST AND AS FOR THAT YOU SHOULD NOT BE SATISFIED WITH IT YET.** Since we know now what's the fishy part, and given all the fresh knowledge in your head, it's much easier to act now than in the future.

Reply:
Your right the stop condition for ScanError is never trigger and the test ends by timeout.
However this was by design, we felt this was a good way of confirming that there was no hidden ScanErrors.

Would you accept a comment explaining this test?

Reply:
It's terrible practice guys. Did you notice the time-out's length? It's terribly long, expressing that it's not the valid path of your tests. We should be all woken up already if a unit tests demand 1 second or more, here you wait 10 endless seconds for a time-out from a SystemKiller. This should always be considered a failure. 

Actually that's a good idea. I'll make a card which will ensure that killing the system by the SystemKiller will never be accepted peacefully as the ending of the test. It should panic so that there are no doubts about it.  
The SystemKiller was invented to help us cope with hanging tests. As long as we can stop them we don't care if we did it by a panic or anything else. 

I first wanted to give you a task of funding a rigid solution without the inflated time-outs. It's not easy actually and I was going to give you some hints but then I realized it was getting quite complicated, beyond what I would be able to precisely describe, at least not at the moment, maybe in another occasion. I tried to rough it out just in thoughts and it let me near an interesting concept that we might use repetitively. 

That was when I realized that the best next step would be to insert it into a new card's description and leave it for some time later, as until then your test is going to slow down our CI quite remarkably but we can live with this awhile.

I'll try to convince Kauri to assign me to this new card I'm going to write with this in mind. 

You please just add a comment referencing the card GH-555. This is going to be one of those tests to be made swifter, as I'm positive that our codebase has already been contaminated with at least one test with such bad effects (discovered indirectly just from observing some tests running long suspiciously). Otherwise, this conversation has ended, I suppose.